connector_acp/models/etd_mixin.py

```diff
@@ -52,74 +54,95 @@ def get_etd_document(self):
         return res
 
     def prepare_keywords(self):
-        """
-        Returns a dictionary of keywords used in the template
+        """Return a dictionary of keywords used in the template.
+
         :return: Dictionary of keywords used in the template
         """
-        return {}
+        return {
+            'o': self,
+            'now': fields.datetime.now(),
+            'today': fields.datetime.today()
+        }
 
     def get_etd_filename(self, file):
         return '%s.%s' % (self.name or self.number, file.file_type)
 
+    def get_etd_directory(self, file, file_list, res_content):
+        filename = self.get_etd_filename(file)
+        for file_dict in file_list:
+            if filename in file_dict:
+                file_dict['content'] += res_content
+
     def build_file(self, file):
-        """
```

Comment:
Add optional argument files. 
If the filename generated for the current record exists in the files, the file template has the grouped flag set, then append content text to it.